CONTRIBUTING.md

```diff
@@ -0,0 +1,127 @@
+# Contributing
+
+## Developer Guide
+
+### Local Development
+
+To spin up a local development environment with locally built artifacts, run:
+
+```
+make build local-dev
+```
+
+You can override default local development configuration by overriding environment
+variables [here](https://github.com/upbound/universal-crossplane/blob/main/cluster/local/config/config.env).
+
+For example, the following will enable connecting your local development environment to Upbound Cloud:
+
+```
+export LOCALDEV_CONNECT_CP_ORG=<YOUR_UBC_ORG>
+export LOCALDEV_CONNECT_API_TOKEN=<YOUR_ACCESS_TOKEN>
+
+make build local-dev
+```
+
+### Cleanup
+
+To clean up local dev environment, first delete self hosted control plane (if connected) from Upbound Cloud Console
+and then run:
+
+```
+make local.down
+```
+
+### Validation
+
+To run validation tests locally, run:
+
+```
+make e2e.run
+```
+
+To run validation test including Upbound Cloud connectivity, run:
+
+```
+export LOCALDEV_CONNECT_CP_ORG=<YOUR_UBC_ORG>
+export LOCALDEV_CONNECT_API_TOKEN=<YOUR_ACCESS_TOKEN>
+
+make e2e.run
+```
+
+## Release Process
+
+A UXP release is cut by following the steps below:
+
+1. **branch repo**: Create a new release branch using the GitHub UI for the
+   repo.
+1. **release branch prep**: Make any release-specific updates on the release
```

Comment:
@muvaf (and cc: @turkenh ) for the UBC components we generate a RELEASE.TXT and commit it so that the tag built on the release branch does not exist on master.  

I'm not seeing the same on crossplane branches, so it looks like this is UBC-only. Thanks!